quota/etcd/quotaapi/quota_server.go

```diff
@@ -200,3 +201,13 @@ func findByName(name string, cfgs *storagepb.Configs) (*storagepb.Config, bool)
 	}
 	return nil, false
 }
+
+func validateName(name string) error {
+	switch {
+	case name == "":
+		return status.Errorf(codes.InvalidArgument, "name is required")
+	case !storage.IsNameValid(name):
```

Comment:
I've made config names mandated by storage, so it's easier to reason that a bad name can't get in. For convenience, those match the names exposed by the API, akin to having a primary key on Config.name. Since the names are storage-mandated, it seemed OK to expose the validation logic from it.

(I hope that answers your question.)